R/stackingutilities.R

```diff
@@ -492,9 +492,241 @@ getMultiOutcomeNames <- function(data)
+#' the information about code B.
+#' @param reduction.list List that contains two elements, \itemize{
+#'  \item nets Character vector of net(s) that exhibit this situation
+#'  \item variable.type Character vector stating the variable set structure that this applies to.
+#' e.g. The predictor grid or outcome variables. Could be a single element or vector with two elements
+#' if both outcome and predictor grids are affected.
+#' }
+#' @noRd
+throwCodeReductionWarning <- function(reduction.list)
+{
+    nets <- reduction.list[["nets"]]
+    variable.type <- reduction.list[["variable.type"]]
+    net.name.txt <- paste0(sQuote(nets, q = FALSE))
+    if (length(nets) > 1)
+        net.name.txt <- paste0(": (", paste0(net.name.txt, collapse = "; "), ")")
+    net.txt <- ngettext(length(nets), "a NET, ", "NETs")
+    variable.type.and <- paste0(paste0(variable.type, collapse = " and "), " variables")
+    variable.type.or <- paste0(paste0(variable.type, collapse = " or "), " variables")
+    net.txt.2 <- ngettext(length(nets), "this NET was", "these NETs were")
+    net.txt.3 <- ngettext(length(nets), "this NET", "any of these NETs")
+    warning("NETs are removed from this analysis unless all their source values are mutually exclusive ",
+            "to other codes. The ", variable.type.and, " have ", net.txt, net.name.txt, " that contains ",
+            "source values that partially overlap with other codes. Consequently, ", net.txt.2, " not used in ",
+            "the analysis. If you wish ", net.txt.3, " to be used in the analysis then please modify the ",
+            variable.type.or, " via the Table view options appropriately.")
+}
+
+#' Stacks several text variables and an existing categorization 
+#' in a consisten manner for text analysis functions.
+#' 
+#' @param text A data frame containing one or more character vectors. Typically
+#' these start their life as variables in a Text - Multi variable set in Displayr
+#' or are several related Text variable sets.
+#' @param existing.categorization A data frame containing the corresponding
+#' categories. This could be a data frame of factor variables (a single-response 
+#' categorization), or a data frame of binary numeric variables (a multiple-response
+#' categorization). This can be NULL if no existing categorization is to be used
+#' in the analysis.
+#' @param subset A boolean vector indicating the subset of cases that the user
+#' wishes to include later in a classification of the text. Usually comes from QFilter.
+#' This will be expanded to align with the stacked data.
+#' @param weights A numeric vector of case weights for later us in the analysis.
+#' This will be expanded to align with the stacked data.
+#' @importFrom utils stack
+#' @importFrom flipFormat TidyLabels ExtractCommonPrefix
+#' @export
+StackTextAndCategorization <- function(text, existing.categorization = NULL, subset = TRUE, weights = NULL) {
+    
```

Comment:
minor improvement. Might make the code below easier to read with the following definition
`has.existing.categorization <- !is.null(existing.categorization)`

Reply:
Be easier to read I think if the checks for valid categorization are done early and error early if invalid. The definitions below can also update the other `attr` calls later to `question.type` and make it easier to read.

e.g.
```suggestion
has.existing.categorization <- !is.null(existing.categorization)
single.text.variable <- is.list(text) && length(text) == 1L
question.type <- if (has.existing.categorization) attr(existing.categorization, "questiontype")
if (has.existing.categorization) {
    if (single.text.variable && !question.type %in% c("PickOne", "PickAny"))
        stop("The existing categorization should be a Nominal/Ordinal or Binary - Multi variable set")
    if (!single.text.variable && !question.type %in% c("PickOneMulti", "PickAnyGrid") 
        stop("The existing categorization should be a Nominal/Ordinal - Multi or Binary - Grid variable set")
}
if (single.text.variable) 
     return(list(text = text[[1]],
                    existing.categorization = existing.categorization,
                    subset = subset,
                    weights = weights))
```